src/coreclr/debug/ee/controller.cpp

```diff
@@ -9054,6 +9054,23 @@ bool DebuggerDataBreakpoint::TriggerSingleStep(Thread *thread, const BYTE *ip)
         this->DisableSingleStep();
         return true;
     }
+    else if (g_pDebugger->m_isSuspendedForGarbageCollection)
+    {
+        // The debugger is not interested in Data Breakpoints during garbage collection
+        // We can safely ignore them since the Data Breakpoints are now on pinned objects
+        LOG((LF_CORDB, LL_INFO10000, "D:DDBP: Ignoring data breakpoint while in SuspendEE() for GC \n"));
```

Comment:
Technically, we need to ignore it not just during `SuspendEE()`, but throughout the whole suspension.

```suggestion
        LOG((LF_CORDB, LL_INFO10000, "D:DDBP: Ignoring data breakpoint while in the runtime is suspended for GC \n"));
```